sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/xml/StaxXmlParser.scala

```diff
@@ -423,19 +396,22 @@ class StaxXmlParser(
                 }
               } else {
                 StaxXmlParserUtils.skipChildren(parser)
+                StaxXmlParserUtils.consumeNextEndElement(parser)
               }
           }
         } catch {
           case e: SparkUpgradeException => throw e
           case NonFatal(e) =>
             badRecordException = badRecordException.orElse(Some(e))
+            StaxXmlParserUtils.skipChildren(parser)
+            StaxXmlParserUtils.consumeNextEndElement(parser)
```

Comment:
This change is not needed. 
This may be a parsing exception. Parsing any further will likely result in more exception. And `badRecordException ` will be lost.

Reply:
Discussed offline. The current implementation doesn't support partial results if there's a bad record exception and I will remove these lines.